src/components/Appointments/MyAppointments.js

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getAppointments, deleteAppointment } from '../../redux/appointments/actions/appointmentActions';
+
+const MyAppointments = () => {
+  const dispatch = useDispatch();
+  const User = useSelector((state) => state.sessionsReducer.fetchedData.user);
+  const appointments = useSelector((state) => state.appointmentReducer.appointments);
+
+  const selfAppoitments = () => {
+    const results = [];
+    if (appointments.length === 0) {
+      appointments.map((appoint) => {
+        if (appoint.user_id === User.id) {
+          results.push(appoint);
+        }
+        return null;
+      });
+    }
+    return results;
+  };
+
+  useEffect(() => {
+    dispatch(getAppointments());
+    selfAppoitments();
+  }, []);
+
+  const handleDelete = (id) => {
+    dispatch(deleteAppointment(id));
+  };
+
+  const myAppointments = selfAppoitments();
+
+  return (
+    <section>
+      <h1>{User.name}</h1>
+      {
+      myAppointments.length === 0 ? (
+        <div>
+          <h2>No Appointments Available</h2>
+        </div>
```

Comment:
nice idea returning  h2 response  but this could be better if you returned it conditionary